### PR TITLE
feat(components): add inline tabs component with ajax support

### PR DESCRIPTION
**Pull request by hypeJunction**
Inline tabs component can now be rendered with "page/components/tabs" view.
The component allows to switch between inline and ajax tabs without reloading
the page

### CODE REVIEW

**Comment by hypeJunction** [comment]
@jdalsem @jeabakker This should help with group tools edit as well.

**Comment by mrclay** [comment]
Does the Ajax-loaded content persist or is it re-requested every time the frame is displayed? Id prefer the latter and control that with HTTP Expires.

**Comment by mrclay** [comment]
> switch between inline and ajax tabs

I was confused by this wording. Maybe "switch between pre-populated or ajax-loaded tabs"?

**Comment by hypeJunction** [comment]
Ajax loaded content persists, but we can expire it - how do I access expiry stamp with the Ajax API?
Sure, firing an event seems useful - we could do it for non-ajax tabs as well.

**Comment by mrclay** [comment]
Yes, same event for all tabs. How about we reload ajax content each time and tackle caching later? It's a hard problem.

**Comment by jdalsem** [comment]
> Yes, same event for all tabs. How about we reload ajax content each time and tackle caching later? It's a hard problem.

Why is this hard? If there already is content for the tab, do not reload... otherwise load from url.
This could be made configurable... if "data-ajax-refresh" is true... always reload... otherwise only if data is missing.

**Comment by jdalsem** [comment]
I think persistent tab content is required to be useful in complex forms..

**Comment by hypeJunction** [comment]
I strongly discourage using partial forms via ajax tabs. Asking for trouble.

**Comment by hypeJunction** [comment]
I will go with the simplest option for now - we can introduce persistent tabs later.

**Comment by mrclay** [comment]
If used on a form you want persistence, if used to display (eg) a river you want fresh content. Straw man: Rather than have a default let's require data-ajax-reload (bool) for all Ajax tabs. Eliminates the surprise.

**Comment by hypeJunction** [comment]
Updated this to:
- use `data-ajax-reload` to control reloading
- use `data-ajax-query` to add custom query elements to the request
- use `data-ajax-href` to set a different URL for ajax requests

File: views/default/page/components/tabs.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		$class[] = $selected ? 'elgg-state-active' : 'hidden';
+
+		$content .= elgg_format_element('div', [
+			'class' => $class,
+			'id' => "{$id}-{$index}",
+		], $tab['content']);
+		unset($tab['content']);
+
+		$tab['href'] = "#{$id}-{$index}";
+	}
```
Comment (mrclay) [inline views/default/page/components/tabs.php]: If `content` not given, `href` should be required.
If `href` given, `data-ajax-reload` should be required.
log error if these aren't met.

File: mod/developers/views/default/theme_sandbox/components/tabs.php [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		'inline' => [
+			'text' => 'Inline Content',
+			'content' => elgg_view('developers/ipsum'),
+		],
+		'ajax' => [
+			'text' => 'Ajax [selected]',
+			'href' => 'ajax/view/theme_sandbox/components/tabs/ajax',
+			'selected' => true,
+			'data-ajax-query' => json_encode([
+				'content' => 'This tab was preselected and loaded via ajax',
```
Comment (mrclay) [inline mod/developers/views/default/theme_sandbox/components/tabs.php]: Demo makes it look like these aren't query string args, but I'm OK with it.

File: views/default/page/components/tabs.php (hunk 1 above)
Comment (hypeJunction) [inline views/default/page/components/tabs.php]: Done

File: mod/developers/views/default/theme_sandbox/components/tabs.php (hunk 2 above)
Comment (hypeJunction) [inline mod/developers/views/default/theme_sandbox/components/tabs.php]: Let's leave for now

File: views/default/page/components/tabs.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			'id' => "{$id}-{$index}",
+		], $tab['content']);
+		unset($tab['content']);
+
+		$tab['href'] = "#{$id}-{$index}";
+	} else {
+		if (!isset($tab['data-ajax-reload'])) {
+			elgg_log('Tab configuration with a "href" parameter should 
+				also specify "data-ajax-reload" parameter.', 'WARNING');
+			$tab['data-ajax-reload'] = true;
```
Comment (mrclay) [inline views/default/page/components/tabs.php]: `continue` here too so devs can't rely on default behavior.

Comment (hypeJunction) [inline views/default/page/components/tabs.php]: You want to be so strict about it?

Comment (mrclay) [inline views/default/page/components/tabs.php]: If it's not really required, just document the default and lose the error. I'm ok either way.

Comment (hypeJunction) [inline views/default/page/components/tabs.php]: Done

**Pull request merged**